File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        N = len(nums)

        lis_length = [1] * N

        for i in range(N):
            for j in range(i):
                if nums[i] > nums[j]:
                    lis_length[i] = max(lis_length[i], lis_length[j] + 1) 

        return max(lis_length) 

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n,1);
        for (int right = 1; right < n; right++) {
            for (int left = 0; left < right; left++) {
                if (nums[left] < nums[right]) {
                    dp[right] = max(dp[right], dp[left] + 1);
                }
            }
        }
        int ans = 0;
        for (int c : dp) {
            ans = max(ans, c);
        }
        return ans;
    }
};